hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/OMDBInsightEndpoint.java

```diff
@@ -0,0 +1,405 @@
+                  keyEntityInfo.getSize());
+          openKeyInsightInfo.setReplicatedTotal(
+              openKeyInsightInfo.getReplicatedTotal() +
+                  keyEntityInfo.getReplicatedSize());
+          boolean added =
+              isLegacyBucketLayout ? nonFSOKeyInfoList.add(keyEntityInfo) :
+                  fsoKeyInfoList.add(keyEntityInfo);
+          if ((nonFSOKeyInfoList.size() + fsoKeyInfoList.size()) == limit) {
+            recordsFetchedLimitReached = true;
+            break;
+          }
+        }
+      } catch (IOException ex) {
+        throw new WebApplicationException(ex,
+            Response.Status.INTERNAL_SERVER_ERROR);
+      } catch (IllegalArgumentException e) {
+        throw new WebApplicationException(e, Response.Status.BAD_REQUEST);
+      } catch (Exception ex) {
+        throw new WebApplicationException(ex,
+            Response.Status.INTERNAL_SERVER_ERROR);
+      }
+      if (recordsFetchedLimitReached) {
+        break;
+      }
+    }
+    return Response.ok(openKeyInsightInfo).build();
+  }
+
+  /** This method retrieves set of keys/files/dirs pending for deletion.
+   * Sample API Response:
+   * {
+   *   "replicatedTotal": -1530804718628866300,
+   *   "unreplicatedTotal": -1530804718628866300,
+   *   "deletedkeyinfo": [
+   *     {
+   *       "omKeyInfoList": [
+   *         {
+   *           "metadata": {},
+   *           "objectID": 0,
+   *           "updateID": 0,
+   *           "parentObjectID": 0,
+   *           "volumeName": "sampleVol",
+   *           "bucketName": "bucketOne",
+   *           "keyName": "key_one",
+   *           "dataSize": -1530804718628866300,
+   *           "keyLocationVersions": [],
+   *           "creationTime": 0,
+   *           "modificationTime": 0,
+   *           "replicationConfig": {
+   *             "replicationFactor": "ONE",
+   *             "requiredNodes": 1,
+   *             "replicationType": "STANDALONE"
+   *           },
+   *           "fileChecksum": null,
+   *           "fileName": "key_one",
+   *           "acls": [],
+   *           "path": "0/key_one",
+   *           "file": false,
+   *           "latestVersionLocations": null,
+   *           "replicatedSize": -1530804718628866300,
+   *           "fileEncryptionInfo": null,
+   *           "objectInfo": "OMKeyInfo{volume='sampleVol', bucket='bucketOne',
+   *           key='key_one', dataSize='-1530804718628866186', creationTime='0',
+   *           objectID='0', parentID='0', replication='STANDALONE/ONE',
+   *           fileChecksum='null}",
+   *           "updateIDset": false
+   *         }
+   *       ]
+   *     }
+   *   ],
+   *   "status": "OK"
+   * }
+   */
+  @GET
+  @Path("/deletePending")
+  public Response getDeletedKeyInfo(
+      @DefaultValue(DEFAULT_FETCH_COUNT) @QueryParam(RECON_QUERY_LIMIT)
+      int limit,
+      @DefaultValue(StringUtils.EMPTY) @QueryParam(RECON_QUERY_PREVKEY)
+      String prevKeyPrefix) {
```

Comment:
Since query to different information, key Info and dir Info, do need have separate information for query separately OR it will be in sequence after finish dir and then query key?

Reply:
> Since query to different information, key Info and dir Info, do need have separate information for query separately OR it will be in sequence after finish dir and then query key?

Since we need to provide only openKeys (keys/files) , so querying only from openKeyTable and openFileTable.

Reply:
> Since query to different information, key Info and dir Info, do need have separate information for query separately OR it will be in sequence after finish dir and then query key?

It will be in sequence.